src/smtsolvers/LookaheadSMTSolver.cc

```diff
@@ -239,120 +457,133 @@ std::pair<LookaheadSMTSolver::laresult,Lit> LookaheadSMTSolver::lookaheadLoop()
-            continue; // Skip the vars that the logic considers bad to split on
-        }
+                continue;
+            }
+            if (v == (idx * nVars()) && skipped_vars_due_to_logic > 0) {
+                respect_logic_partitioning_hints = false; // Allow branching on these since we looped back.
+            }
+            if (respect_logic_partitioning_hints && !okToPartition(v)) {
+                skipped_vars_due_to_logic++;
+                std::cout << "Skipping " << v << " since logic says it's not good\n";
+                continue; // Skip the vars that the logic considers bad to split on
+            }
 #ifdef LADEBUG
-        printf("Checking var %d\n", v);
+                printf("Checking var %d\n", v);
 #endif
-        Lit best = score->getBest();
-        if (value(v) != l_Undef || (best != lit_Undef && score->safeToSkip(v, best))) {
+            Lit best = score->getBest();
+            if (value(v) != l_Undef or (best != lit_Undef and score->safeToSkip(v, best))) {
 #ifdef LADEBUG
-            printf("  Var is safe to skip due to %s\n",
-                   value(v) != l_Undef ? "being assigned" : "having low upper bound");
+                printf("  Var is safe to skip due to %s\n",
+                       value(v) != l_Undef ? "being assigned" : "having low upper bound");
 #endif
-            score->setChecked(v);
-            // It is possible that all variables are assigned here.
-            // In this case it seems that we have a satisfying assignment.
-            // This is in fact a debug check
-            if (static_cast<unsigned int>(trail.size()) == dec_vars) {
+                if (value(v) != l_Undef and next_arr[v]) {
+                    next_arr[v] = false;
+                    close_to_prop--;
+                }
+                // It is possible that all variables are assigned here.
+                // In this case it seems that we have a satisfying assignment.
+                // This is in fact a debug check
+                if (static_cast<unsigned int>(trail.size()) == dec_vars) {
 #ifdef LADEBUG
-                printf("All vars set?\n");
+                    printf("All vars set?\n");
 #endif
-                if (checkTheory(true) != TPropRes::Decide)
-                    return {laresult::la_tl_unsat, lit_Undef}; // Problem is trivially unsat
-                assert(checkTheory(true) == TPropRes::Decide);
+
+                    if (checkTheory(true) != TPropRes::Decide)
+                        return {laresult::la_tl_unsat, best}; // Problem is trivially unsat
+                    assert(checkTheory(true) == TPropRes::Decide);
 #ifndef NDEBUG
-                for (CRef cr : clauses) {
-                    Clause& c = ca[cr];
-                    unsigned k;
-                    for (k = 0; k < c.size(); k++) {
-                        if (value(c[k]) == l_True) {
-                            break;
+                    for (int j = 0; j < clauses.size(); j++) {
```

Comment:
Any reason why this has been turned from range-based for loop to classic for-loop? I don't see it.